sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp

```diff
@@ -163,6 +163,49 @@ namespace Azure { namespace Storage {
         std::string LeaseId;
       };
 
+      /**
+       * @brief An Azure Storage blob.
+       */
+      struct BlobItem final
+      {
+        /**
+         * Blob name.
+         */
+        std::string Name;
+        /**
+         * Indicates whether this blob was deleted.
+         */
+        bool IsDeleted = bool();
+        /**
+         * A string value that uniquely identifies a blob snapshot.
+         */
+        std::string Snapshot;
+        /**
+         * A string value that uniquely identifies a blob version.
+         */
+        Nullable<std::string> VersionId;
+        /**
+         * Indicates if this is the current version of the blob.
+         */
+        Nullable<bool> IsCurrentVersion;
+        /**
+         * Properties of a blob.
+         */
+        BlobItemDetails Details;
+        /**
+         * Indicates that this root blob has been deleted, but it has versions that are active.
+         */
+        Nullable<bool> HasVersionsOnly;
+        /**
+         * Size in bytes.
+         */
+        int64_t BlobSize = int64_t();
```

Comment:
Initializer form instead of assigning to constructor.

```suggestion
        int64_t BlobSize{};
```